src/sentry/static/sentry/app/components/charts/worldMapChart.tsx

```diff
@@ -0,0 +1,159 @@
+import {withTheme} from 'emotion-theming';
+import max from 'lodash/max';
+import PropTypes from 'prop-types';
+import React from 'react';
+import echarts, {EChartOption} from 'echarts';
+
+import {Series, SeriesDataUnit} from 'app/types/echarts';
+import theme from 'app/utils/theme';
+
+import BaseChart from './baseChart';
+import MapSeries from './series/mapSeries';
+import VisualMap from './components/visualMap';
+
+type ChartProps = React.ComponentProps<typeof BaseChart>;
+
+type MapChartSeriesDataUnit = Omit<SeriesDataUnit, 'name' | 'itemStyle'> & {
+  name: string | undefined;
+  // Docs for map itemStyle differ from Series data unit. See https://echarts.apache.org/en/option.html#series-map.data.itemStyle
+  itemStyle: EChartOption.SeriesMap.DataObject['itemStyle'];
+};
+
+type MapChartSeries = Omit<Series, 'data'> & {
+  data: MapChartSeriesDataUnit[];
+};
+
+type Props = Omit<ChartProps, 'series'> & {
+  series: MapChartSeries[];
+  seriesOptions?: EChartOption.SeriesMap;
+};
+
+type JSONResult = {[key: string]: any};
+
+type State = {
+  countryToCodeMap: JSONResult | null;
+  map: JSONResult | null;
+  codeToCountryMap: JSONResult | null;
+};
+
+class WorldMapChart extends React.Component<Props, State> {
+  static propTypes = {
+    ...BaseChart.propTypes,
+    seriesOptions: PropTypes.object,
+  };
+
+  state = {
```

Comment:
```suggestion
  state: State = {
```

sometimes I need to add the State type here as well. Please does it work fine for you without it?

Reply:
I like to add it regardless, just missed this one.